core/src/main/java/cz/xtf/core/config/OpenShiftConfig.java

```diff
@@ -48,8 +68,44 @@ public static String version() {
         return XTFConfig.get(OPENSHIFT_VERSION);
     }
 
+    /**
+     * @return Returns default namespace as defined in xtf.openshift.namespace property or namespace for currently running test
+     *         case when:
+     *         -Dxtf.openshift.namespace.per.testcase=true.
+     *         In case when current thread does not have associated test case (for example when initializing Openshift instance
+     *         in static variable or static block) then exception is thrown.
+     */
     public static String namespace() {
-        return XTFConfig.get(OPENSHIFT_NAMESPACE);
+        if (useNamespacePerTestcase()) {
```

Comment:
still valid

Reply:
Thinking of it, It would introduce another entry point where namespace can be retrieved. I wanted to have just one place where this information could be retrieved. Now it's `OpenShiftConfig.namespace()`. I don't think there is a room to change it as moving it to `NamespaceManager.getNamespace()` would require refactoring/changes in consuming test suite.

Reply:
I guess we'll need to make this change sooner or later. However we need to behave correctly like if we're at 1.0 version as XTF is consumed by number of teams and they would not be happy if anything breaks. 

What about to deprecate `OpenShiftConfig.namespace()` and use `NamespaceManager.getNamespace()` instead. And make moving consuming TSs to `NamespaceManager.getNamespace()` are prerequisite for this feature.

Reply:
> What about to deprecate `OpenShiftConfig.namespace()` and use `NamespaceManager.getNamespace()` instead. And make moving consuming TSs to `NamespaceManager.getNamespace()` are prerequisite for this feature.

I agree.